Infinite scroll of posts

motuncoded opened the pull request:
In this change, I created a Posts.tsx file that can allow infinite scroll of posts with loading when navigating down the posts.

closes #5

Commit:
Create a simple scroll of posts

In this feature, I used Posts Placeholder Api which  render a simple scroll of different posts.

Commit:
Setup the prettier and made file changes

In this change, I set up prettier in the project and move the function types outside the Reactjs function

Commit:
Create an infinite scroll of posts

In this change, I created a Posts.tsx file that can allow infinite scroll of posts with loading when navigatimg down the posts.

Commit:
Create an infinite scroll of posts

In this change, I created a Posts.tsx file that can allow infinite scroll of posts with loading when navigatimg down the posts.

motuncoded committed:
Merge branch 'main' into infinite-scroll-of-posts

Asylcreek commented on pages/Posts.tsx:
```diff
@@ -0,0 +1,114 @@
+type Error = {
+  message: string;
+};
+type PageNumber = number;
```
You did not need to create a type alias for this.

Typing line 26 as `number` would have been enough

Asylcreek commented on pages/Posts.tsx:
```diff
@@ -0,0 +1,114 @@
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [page, setPage] = useState<PageNumber>(1);
+  const [hasMore, setHasMore] = useState(true);
```
I don't think `true` should be the default

Asylcreek commented on pages/Posts.tsx:
```diff
@@ -0,0 +1,114 @@
+      }
+      const posts = await response.json();
+      setPosts((prev)=> [...prev, ...posts]);
+      if (posts.length < 10) {
+        setHasMore(false);
+      }
```
Flip this condition around

Asylcreek commented on pages/Posts.tsx:
```diff
@@ -0,0 +1,114 @@
+
+
+  const handleScroll = () => {
+    if (window.innerHeight + document.documentElement.scrollTop + 1 >=
+      document.documentElement.scrollHeight
```
Shouldn't these be `document.body` instead?

Asylcreek commented on pages/Posts.tsx:
```diff
@@ -0,0 +1,114 @@
+
+
+
+  const handleScroll = () => {
```
Move this function inside the useEffect below

Asylcreek commented on pages/Posts.tsx:
```diff
@@ -0,0 +1,114 @@
+  useEffect(() => {
+    window.addEventListener("scroll", handleScroll);
+    return () => window.removeEventListener("scroll", handleScroll);
+  });
```
```suggestion
  }, []);
```

Commit:
Fix the infinite scroll of posts

In this change I fixed the changes based on the review to have an infinite scroll of posts

Commit:
Merge branch 'infinite-scroll-of-posts' of https://github.com/motuncoded/social_website into infinite-scroll-of-posts

Commit:
optimising next build

In this change I optimised the project by runnning the nex build command to be able to check the vercel preview

Asylcreek commented on pages/Posts.tsx:
Why did you create this in a separate file?

Do you realize that `/` and `/posts` would practically be showing the same things?

motuncoded replied:
ok

motuncoded replied:
I noticed that the posts rendered twice according to id and I have been trying to fix it and it seems unattainable, can you please recommend something
![copt](https://github.com/motuncoded/social_website/assets/67984447/e3472246-1eb0-43c6-868a-43dbbd230112)

Asylcreek replied:
What exactly do you want to happen?

Which route do you want your users to access posts from?

Commit:
adjust the files

In this change, I removed the posts.tsx and incorporated the rendering of the API inside the index.tsx file to avoid rendering the same thing

Commit:
Fix the problem of the loading state

In this change, I made the loading state to render the right jsx elements for loading and no more posts
